foca/api/register_openapi.py

```diff
@@ -0,0 +1,180 @@
+"""Helper class & function definitions for registering OpenAPI specs with a
+Connexion app instance."""
+
+from json import load, decoder
+import logging
+import os
+from typing import Dict, List, Optional
+
+from connexion import App
+from connexion.exceptions import InvalidSpecification
+from yaml import (safe_load,
+                  safe_dump,
+                  parser,
+                  )
+
+from foca.config.config_parser import get_conf
+
+# Get logger instance
+logger = logging.getLogger(__name__)
+
+
+class OpenAPIConfig(Dict):

```

Comment:
For example, to test this class, you could
* import it
* instantiate it with some fake arguments
* verify that the returned object is of type `OpenAPIConfig`

You don't need to test what happens if the user passes an arg that the constructor doesn't define, because if you do that you are testing _Python's_ behavior, not that of your code. You could, however, write code that validates the user's params (are they of the expected type) and raise descriptive errors accordingly, to help the user use the code in the right way. If you add such code, you could write tests with fake args that are of the wrong type.